### PR TITLE
feat: TET-858 boolean pill

### PR DESCRIPTION
**Pull request by golas-m**
Added the boolean pill component
[Figma design](https://www.figma.com/design/4kcoCOyXVI4TIKAh60P44C/Tetrisly-Design-System-v.1.1.0?node-id=3420-42972&t=LIiPpCUH3og1AroH-0)

### CODE REVIEW

**Comment by adrian-potepa** [comment]
The `padding-left` value is wrong for all `beforeComponent` prop values:

Zrzut ekranu 2024-05-9 o 16 29 35

Here is image from UI:

Zrzut ekranu 2024-05-9 o 16 28 41

File: src/components/BooleanPill/BooleanPill.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    if (state !== 'disabled') {
+      onChange?.(state === 'default');
+    }
+  }, [onChange, state]);
+
+  return (
+    <tet.span
+      data-state={state}
+      onClick={handleOnClick}
+      data-testid="boolean-pill"
```
Comment (adrian-potepa) [inline src/components/BooleanPill/BooleanPill.tsx:50]: Tou should add here some default `tabIndex` value to handle `:focus` and `:active` states.

File: src/components/BooleanPill/BooleanPill.styles.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  >;
+} & BaseProps;
+
+export const defaultConfig = {
+  display: 'inline-flex',
+  justifyContent: 'center',
+  alignItems: 'center',
+  textAlign: 'center',
+  whiteSpace: 'nowrap',
+  minWidth: `62px`,
```
Comment (adrian-potepa) [inline src/components/BooleanPill/BooleanPill.styles.ts:17]: It's not necessary I guess 🤔

File: src/components/BooleanPill/BooleanPill.props.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+export type BooleanPillProps = {
+  text: string;
+  state?: 'default' | 'disabled' | 'selected';
+  isInverted?: boolean;
+  custom?: BooleanPillConfig;
+  avatar?:
+    | { appearance?: 'image'; image: string }
+    | { appearance: Exclude<AvatarAppearance, 'image'>; initials: string };
+  onChange?: (state: boolean) => void;
```
Comment (adrian-potepa) [inline src/components/BooleanPill/BooleanPill.props.ts:12]: I think, we should allow here to support all native react synthetic events `onClick`, 'onFocus', `onBlur` etc.
You can add here optional `tabIndex` prop type which will be needed to support `:focus` state in css.

File: src/components/BooleanPill/BooleanPill.props.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { BooleanPillConfig } from './BooleanPill.styles';
+import { AvatarAppearance } from '../Avatar/types';
+
+export type BooleanPillProps = {
+  text: string;
+  state?: 'default' | 'disabled' | 'selected';
```
Comment (adrian-potepa) [inline src/components/BooleanPill/BooleanPill.props.ts:6]: `selected` should be as `boolean` props `isSelected`, same as `isInverted`.
In `state` props are states which cant be mixed with other one, we can set `active` and `disabled` in the same time ;)

**Review by adrian-potepa — CHANGES_REQUESTED** [review]
Good job! Only few small issues to work with.

File: src/components/BooleanPill/BooleanPill.styles.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  >;
+} & BaseProps;
+
+export const defaultConfig = {
+  display: 'inline-flex',
+  justifyContent: 'center',
+  alignItems: 'center',
+  textAlign: 'center',
+  whiteSpace: 'nowrap',
+  h: `$size-small`,
```
Comment (mwleklinski) [inline src/components/BooleanPill/BooleanPill.styles.ts:19]: no need to use \`\` you can use "" instead

**Review by mwleklinski — APPROVED** [review]

**Comment by adrian-potepa** [comment]
I noticed `ker` error with story:
image

**Review by adrian-potepa — CHANGES_REQUESTED** [review]
`key` error in stories

**Review by adrian-potepa — APPROVED** [review]

**Pull request merged**